### PR TITLE
feat(ci): add canary

### PR DESCRIPTION
**Pull request by cartermckinnon**
**Description of changes:**

This refactors the bot-driven CI workflow to be reusable by a new time-driven "canary" workflow.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          additional_arguments: ${{ inputs.build_arguments }}
+      - if: ${{ inputs.goal == 'test' }}
+        name: "${{ needs.setup.outputs.ci_step_name_prefix }} Test"
+        id: test
+        uses: ./.github/actions/ci/kubetest2
+        with:
+          ami_id: ${{ steps.build.outputs.ami_id }}
+          k8s_version: ${{ matrix.k8s_version }}
+          os_distro: ${{ matrix.os_distro }}
+          build_id: ${{ needs.setup.outputs.build_id }}
```
Comment (ndbaker1) [inline .github/workflows/ci.yaml:91]:
```suggestion
          build_id: ${{ inputs.resource_id }}
```

File: .github/workflows/ci-manual.yaml [hunk 2]
```diff
@@ -141,16 +103,19 @@ jobs:
             };
             const uniqueStepNames = new Set();
             const stepConclusionsByJob = new Map();
-            const ciStepNamePrefix = "${{ needs.setup.outputs.ci_step_name_prefix }}";
+            const ciStepNamePrefix = "${{ needs.ci.outputs.ci_step_name_prefix }}";
             for (const job of data.jobs) {
-              if (/^\d+\.\d+/.test(job.name)) {
+              // look for jobs named "ci / k8sVersion / osDistro"
+              if (/^ci \/ \d+\.\d+/.test(job.name)) {
```
Comment (ndbaker1) [inline .github/workflows/ci-manual.yaml:109]: this didn't change did it? the step name definition is still
```
    name: ${{ matrix.k8s_version }} / ${{ matrix.os_distro }}
```

Comment (cartermckinnon) [reply .github/workflows/ci-manual.yaml:109]: GitHub prefixes the step name with the reusable workflows name

Comment (ndbaker1) [reply .github/workflows/ci-manual.yaml:109]: ah thanks, the regex and trimming feels sorta awkward but i tried perusing the api workflows response (https://api.github.com/repos/cartermckinnon/amazon-eks-ami/actions/runs/11411576922/jobs) and i can't really come up with a better plan either 🤷

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        type: string
+        default: "test"
+      os_distros:
+        description: 'Operating System Distributions (comma-separated, e.g., al2,al2023)'
+        default: "al2,al2023"
+        required: false
+        type: string
+      k8s_versions:
+        description: 'Kubernetes Versions (comma-separated, e.g., 1.29,1.30)'
+        default: "1.24,1.25,1.26,1.27,1.28,1.29,1.30,1.31"
```
Comment (ndbaker1) [inline .github/workflows/ci.yaml:26]: any easily way to get rid of the duplicate parameters in here and `ci-manual`?

**Review by ndbaker1 — APPROVED** [review]

File: .github/workflows/ci-manual.yaml (hunk 2 above)
Comment (cartermckinnon) [reply .github/workflows/ci-manual.yaml:109]: Yeah it gets the job done for now

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (cartermckinnon) [reply .github/workflows/ci.yaml:26]: I think they can be removed from the bot-driven workflow. I’ll open a follow up to do some renaming and cleanup

**Pull request merged**